core/templates/dev/head/pages/exploration_editor/editor_tab/StateInteraction.js

```diff
@@ -262,6 +262,15 @@ oppia.controller('StateInteraction', [
                   });
                 }
 
+                if (Object.keys(
+                  stateCustomizationArgsService.displayed).length === 0) {
+                  $scope.save();
+                  $scope.hasCustomizationArgs = false;
+                } else {
+                  $scope.hasCustomizationArgs = true;
+                }
+                stateInteractionIdService.displayed = newInteractionId;
```

Comment:
I earlier thought of a different solution which required this to be shifted down. Now it need not be shifted below.